### PR TITLE
vault: fix build for darwin

### LINKED ISSUE #129443: vault no longer installs on darwin since 08ab2d3c2abd67b48d3f81f6d77e2fc240b0b4eb
**Describe the bug**

After 08ab2d3c2abd, (#128516) vault will not install on Mac OS.  The backport PR likely also breaks Mac OS.

**To Reproduce**

Build any config including vault on Mac OS.

**Expected behavior**

vault is usable.

**Screenshots**

```
jfelice@C02CW0J5ML87 dotfiles (develop)
$ darwin-rebuild build --flake ./
warning: Git tree '/Users/jfelice/src/dotfiles' is dirty
building the system configuration...
warning: Git tree '/Users/jfelice/src/dotfiles' is dirty
error: Package ‘glibc-2.32-48’ in /nix/store/3xf8klp0rmn6iag9agsfwmzw5rw5g4vd-source/pkgs/development/libraries/glibc/default.nix:149 is not supported on ‘x86_64-darwin’, refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.
(use '--show-trace' to show detailed location information)

```

Further, setting `NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM` does not help.

**Additional context**

**Notify maintainer**

@jonringer @manveru 

**Metadata**

 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.4.0, macOS 10.15.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `"nixpkgs-20.09pre240394.ca3fa9c32a4"`
 - nixpkgs: `/nix/store/8bz4hy23zy32ijh96wz8v3s6mr8ndlnm-source`

Maintainer information:
```yaml
attribute:
- vault
```

### PR DESCRIPTION
**Pull request by eraserhd**
###### Motivation for this change

Closes #129443

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by eraserhd** [comment]
cc @jonringer @manveru

File: pkgs/tools/security/vault/default.nix [hunk 1]
```diff
@@ -26,7 +26,7 @@ buildGoPackage rec {
   postInstall = ''
     echo "complete -C $out/bin/vault vault" > vault.bash
     installShellCompletion vault.bash
-
+  '' + lib.optionalString (!stdenv.isDarwin) ''
```
Comment (jonringer) [inline pkgs/tools/security/vault/default.nix:29]:
```suggestion
  '' + lib.optionalString stdenv.isLinux ''
```

**Review by jonringer — COMMENTED** [review]
do you mind also applying the same fix to `vault-bin`?

File: pkgs/tools/security/vault/default.nix (hunk 1 above)
Comment (eraserhd) [reply pkgs/tools/security/vault/default.nix:29]: will do

**Comment by eraserhd** [comment]
@jonringer changed to use isLinux, and same change made to vault-bin

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129447` at e0cda370 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>vault</li>
    <li>vault-bin</li>
  </ul>
</details>

**Pull request merged**